src/wp-includes/blocks.php

```diff
@@ -134,7 +134,15 @@ function register_block_script_handle( $metadata, $field_name ) {
 		$script_asset['dependencies'],
 		$script_asset['version']
 	);
-	return $result ? $script_handle : false;
+	if ( ! $result ) {
+		return false;
+	}
+
+	if ( ! empty( $metadata['textdomain'] ) ) {
```

Comment:
@ocean90, is it good to assume that if there is no `textdomain` defined that this block/plugin isn't translated?

Reply:
I'd say so, yes. Would core blocks set it to `default`? But it actually doesn't seem like it will ever be empty due to the default being `default`? https://github.com/WordPress/wordpress-develop/blob/b35353aa0a9290589c8906140621831ba5ca0742/src/wp-includes/blocks.php#L238

Reply:
True, I thought I removed this line 😅 
Yes, I believe we either explicitly set it to `default` for core blocks or auto-detect them through `core` namespace in the block name and auto-apply `default` this way.

Reply:
I removed this fallback for now as it's way more complex for blocks than for plugins. 

```php
    // If no text domain is defined fall back to the plugin slug.
    if ( ! $plugin_data['TextDomain'] ) {
        $plugin_slug = dirname( plugin_basename( $plugin_file ) );
        if ( '.' !== $plugin_slug && false === strpos( $plugin_slug, '/' ) ) {
            $plugin_data['TextDomain'] = $plugin_slug;
        }
    }
```

Plugins fall back to the plugin slug, so we could do the same. What do you think?